src/subdomains/monitoring/observers/staking-combined.observer.ts

```diff
@@ -99,4 +106,24 @@ export class StakingCombinedObserver extends MetricObserver<StakingData> {
     const difference = Util.round(actual - should, Config.defaultVolumeDecimal);
     return { actual, should, difference };
   }
+
+  private async getLastOutputDates(): Promise<LastOutputDates> {
+    const lastOutputDates = {};
+
+    for (const strategy in StakingStrategy) {
+      lastOutputDates[StakingStrategy[strategy]] = await this.getLastOutputDate(StakingStrategy[strategy]);
+    }
+
+    return lastOutputDates as LastOutputDates;
+  }
+
+  private async getLastOutputDate(strategy: StakingStrategy): Promise<Date> {
+    return await getCustomRepository(RewardRepository)
+      .findOne({
+        order: { reinvestOutputDate: 'DESC' },
+        where: { staking: { strategy } },
+        relations: ['staking'],
+      })
+      .then((b) => b.reinvestOutputDate);
+  }
```

Comment:
Define the type with `?` (date might be undefined)
```ts
type LastOutputDates = {
  [k in StakingStrategy]?: Date;
};
```
then define the variable like
```ts
const lastOutputDates: LastOutputDates = {};
```
return it without cast
```ts
return lastOutputDates;
```
and add a `?` in the then
```ts
.then((b) => b?.reinvestOutputDate);
```